sympy/functions/elementary/piecewise.py

```diff
@@ -1230,3 +1168,87 @@ def _clip(A, B, k):
         pass
 
     return p
+
+
+def piecewise_simplify_arguments(expr, **kwargs):
+    from sympy import simplify
+    args = []
+    for e, c in expr.args:
+        if isinstance(e, Basic):
+            olddoit = kwargs.pop('doit', None)
```

Comment:
Since there will be no confusion between a keyword and variable, I would suggest leaving this `doit=` instead of `olddoit`:
```python
>>> def foo(**k): print(k)
>>> doit=1
>>> foo(doit=doit)
{'doit': 1}
```

Reply:
Not sure if I follow. Here, I would like to not use doit (to avoid growth), but for simplification of the conditions I would like to honor the setting. If I just add doit=False and passes on kwargs, there will be a complain about two doit.

Or you just mean the naming of variable? That can be fixed. Not sure what the idea was.

Reply:
> just mean the naming of variable

yes. I was demponstrating that there is no harm in using variables that match the assumption keyword. No need for cleverness here...move along. :-)